### PR TITLE
feat: store widget localIdentifier in dashboard layout

### PR DESCRIPTION
**Pull request by scavnickyj**
JIRA: F1-523
risk: medium

---

> [!IMPORTANT]
> Please, **don't forget to run `rush change`** for the commits that introduce **new features** 🙏

---

Refer to [documentation](https://github.com/gooddata/gooddata-ui-sdk/blob/master/dev_docs/continuous_integration.md) to see how to run checks and tests in the pull request. This is the list of the most used commands:

```
extended test - backstop
```

```
extended test - tiger-cypress - integrated
extended test - tiger-cypress - isolated
extended test - tiger-cypress - record
```

### CODE REVIEW

**Comment by scavnickyj** [comment]
extended test - tiger-cypress - isolated

**Comment by scavnickyj** [comment]
extended test - tiger-cypress - integrated

**Comment by scavnickyj** [comment]
extended test - tiger-cypress - isolated

**Comment by scavnickyj** [comment]
extended test - tiger-cypress - isolated

**Comment by scavnickyj** [comment]
extended test - tiger-cypress - record insightOnDashboard

**Comment by yenkins** [comment]
# ✨ Cypress test results ✨

## 🎉 Great news! Tests were successful.

|Status|Name|Tests|Failures|Time|
|:--:|:--|:--:|:--:|--:|
|✅|**insightOnDashboard.spec.ts**|9(4 skipped)|-|35.469s|

You can find full report [here](https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tiger-record-zuul/195/).

**Comment by scavnickyj** [comment]
extended test - tiger-cypress - isolated

File: libs/sdk-ui-dashboard/src/_staging/dashboard/dashboardLayout.ts [hunk 1]
```diff
@@ -261,13 +267,22 @@ export function dashboardLayoutRemoveIdentity<T extends IDashboardWidget>(
                     return item;
                 }
 
+                /**
+                 * We want to keep localIdentifier which is the only widget identity stored on backend.
+                 * If it is nonexistent, we create a new one.
+                 */
+                const localIdentifierObj = isWidgetIdentifiersEnabled
+                    ? { localIdentifier: item.widget?.localIdentifier ?? uuidv4() }
+                    : { localIdentifier: undefined };
```
Comment (kandl) [inline libs/sdk-ui-dashboard/src/_staging/dashboard/dashboardLayout.ts:276]: Should this be rather empty, in case it's turned off?

Comment (scavnickyj) [reply libs/sdk-ui-dashboard/src/_staging/dashboard/dashboardLayout.ts:276]: true, will fix that

**Comment by scavnickyj** [comment]
extended test - tiger-cypress - isolated

**Comment by scavnickyj** [comment]
extended test - tiger-cypress - isolated

**Comment by scavnickyj** [comment]
extended test - tiger-cypress - integrated

**Comment by scavnickyj** [comment]
extended test - tiger-cypress - isolated

**Review by kandl — APPROVED** [review]
LGTM

**Pull request merged**